res/qml/WaveformRow.qml

```diff
@@ -0,0 +1,380 @@
+
+                anchors.top: waveform.top
+                anchors.right: waveform.left
+                width: Math.max(0, waveform.x)
+                height: waveform.height
+
+                ShapePath {
+                    strokeColor: Theme.waveformPrerollColor
+                    strokeWidth: 1
+                    fillColor: "transparent"
+
+                    PathMultiline {
+                        paths: {
+                            let p = [];
+                            for (let i = 0; i < preroll.numTriangles; i++) {
+                                p.push([
+                                        Qt.point(preroll.width - i * preroll.triangleWidth, preroll.triangleHeight / 2),
+                                        Qt.point(preroll.width - (i + 1) * preroll.triangleWidth, 0),
+                                        Qt.point(preroll.width - (i + 1) * preroll.triangleWidth, preroll.triangleHeight),
+                                        Qt.point(preroll.width - i * preroll.triangleWidth, preroll.triangleHeight / 2),
+                                ]);
+                            }
+                            return p;
+                        }
+                    }
+                }
+            }
+
+            Shape {
+                id: postroll
+
+                property real triangleHeight: waveform.height
+                property real triangleWidth: 0.25 * waveform.effectiveZoomFactor
+                property int numTriangles: Math.ceil(width / triangleWidth)
+
+                anchors.top: waveform.top
+                anchors.left: waveform.right
+                width: waveformContainer.width / 2
+                height: waveform.height
+
+                ShapePath {
+                    strokeColor: Theme.waveformPostrollColor
+                    strokeWidth: 1
+                    fillColor: "transparent"
+
+                    PathMultiline {
+                        paths: {
+                            let p = [];
+                            for (let i = 0; i < postroll.numTriangles; i++) {
+                                p.push([
+                                        Qt.point(i * postroll.triangleWidth, postroll.triangleHeight / 2),
+                                        Qt.point((i + 1) * postroll.triangleWidth, 0),
+                                        Qt.point((i + 1) * postroll.triangleWidth, postroll.triangleHeight),
+                                        Qt.point(i * postroll.triangleWidth, postroll.triangleHeight / 2),
+                                ]);
+                            }
+                            return p;
+                        }
+                    }
+                }
+            }
+
+            Repeater {
+                model: root.deckPlayer.beatsModel
+
+                Rectangle {
+                    property real alpha: 0.9 // TODO: Make this configurable (i.e., "[Waveform],beatGridAlpha" config option)
```

Comment:
AFAIK, `[Waveform],beatGridAlpha` already exists and is configurable in preferences, any reason for not using it?

Reply:
There is no interface implemented to query these values in QML.